ui/pages/confirmations/hooks/useSyncConfirmPath.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+import { useParams, useLocation } from 'react-router-dom-v5-compat';
+
+import {
+  CONFIRM_TRANSACTION_ROUTE,
+  CONFIRMATION_V_NEXT_ROUTE,
+} from '../../../helpers/constants/routes';
+import { Confirmation } from '../types/confirm';
+import { useConfirmationNavigation } from './useConfirmationNavigation';
+
+const useSyncConfirmPath = (
+  currentConfirmation?: Confirmation,
+  routeParamId?: string,
+) => {
+  const { navigateToId, confirmations } = useConfirmationNavigation();
+  const location = useLocation();
+  const confirmationId = currentConfirmation?.id;
+
+  // Use routeParamId from props if available (passed from v5 Route),
+  // otherwise fall back to useParams() for v5-compat Routes
+  const urlParams = useParams<{ id: string }>();
+  const paramId = routeParamId || urlParams.id;
```

Comment:
Fixed confirmation navigation to start at first confirmation (confirmations[0]) 
and pass ID via props/context since `useParams()` doesn't work in v5 Route render props.

**_TODO for v6 migration:_** 
Remove parameter passing from `useSyncConfirmPath`, 
ConfirmContextProvider, and component props - `useParams()` will work natively 
once `routes.component.tsx` is migrated from react-router-dom v5 to v6.